hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/MemberInfoUpdateOperation.java

```diff
@@ -96,13 +98,24 @@ final void processPartitionState() {
         node.partitionService.processPartitionRuntimeState(partitionRuntimeState);
     }
 
-    protected final boolean isValid() {
+    protected final boolean checkValid() {
         final ClusterServiceImpl clusterService = getService();
         final Connection conn = getConnection();
-        final Address masterAddress = conn != null ? conn.getEndPoint() : null;
+
         boolean isLocal = conn == null;
-        return isLocal
-                || (masterAddress != null && masterAddress.equals(clusterService.getMasterAddress()));
+
+        if (isLocal) {
+            return true;
+        }
+
+        Address endpoint = conn.getEndPoint();
+        Address masterAddress = clusterService.getMasterAddress();
+        boolean valid = (endpoint != null && endpoint.equals(masterAddress));
+        if (!valid) {
+            getLogger().warning("Ignoring operation because sender: " + endpoint + " is not known master: " + masterAddress);
```

Comment:
do we need a warning here? it can happen what a master disconnected while having in-flight operations = business as usual?